pkg/engine/task/task_toggle.go

```diff
@@ -0,0 +1,66 @@
+package task
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	toggleTaskError = "ToggleTaskError"
+)
+
+// ToggleTask will apply or delete a set of given resources to the cluster based on value of Parameter. See Run method for more details.
+type ToggleTask struct {
+	Name      string
+	Parameter string
+	Resources []string
+}
+
+func (tt ToggleTask) Run(ctx Context) (bool, error) {
+	// 1. - Get the task to run
+	task, err := tt.getTask(ctx)
+	if err != nil {
+		return false, fatalExecutionError(err, toggleTaskError, ctx.Meta)
+	}
+	// 2. - Run the returned task
+	return task.Run(ctx)
+}
+
+func (tt ToggleTask) convertToTaskSpec() v1beta1.TaskSpec {
+	return v1beta1.TaskSpec{
+		ResourceTaskSpec: v1beta1.ResourceTaskSpec{
+			Resources: tt.Resources,
+		},
+	}
+}
+
+func (tt ToggleTask) getTask(ctx Context) (Tasker, error) {
+	var task Tasker
+	// 1. - Get the parameter value
+	paramValue := ctx.Parameters[tt.Parameter]
+	if len(paramValue) == 0 {
```

Comment:
You don't need this check, `strconv.ParseBool("")` already gives you `invalid syntax`